Fix HMAC signature calculation

danielduqma opened the pull request:
**Description**
Following issue https://github.com/Adyen/adyen-python-api-library/issues/96, we reviewed HMAC signature calculation, following [Adyen docs](https://docs.adyen.com/development-resources/notifications/verify-hmac-signatures).

**Tested scenarios**
Unit test for updated method has been updated following linked docs, with given example. There are other tests that should be updated too.

**Fixed issue**:  https://github.com/Adyen/adyen-python-api-library/issues/96

Fixes #96

danielduqma commented:
Hi @KadoBOT!
I suppose tests belong to a previous version of Adyen API, so it will take a bit longer to review all of them. I'll update the PR as soon as I can, but any help is welcome.

Thanks.

danielduqma commented:
I corrected failing tests. Travis displays as pending but [seems to be green](https://travis-ci.org/github/Adyen/adyen-python-api-library/builds/670139337?utm_source=github_status&utm_medium=notification). Thanks!

rikterbeek commented:
@danielduqma,

Thanks for the PR! There are two different HMAC calculation one for the Notifications, the one you refer to but as well one for the Hosted Payment Page (HPP). You can find that documentation here: https://docs.adyen.com/classic-integration/hosted-payment-pages/hmac-signature-calculation

The generate_hpp_sig is used for only the HPP flow. If you want to use the HMAC calculation for the notifications could you please change the Pull Request so it is a separate util method for this ? Now this will break all HPP signatures and current integrations.

Regards,
Rik
Adyen

danielduqma commented:
Thanks for the point @rikterbeek. I updated the PR and added two methods for Notifications. Following your comments, I kept tests untouched, and I just added the webhook case described in [docs](https://docs.adyen.com/development-resources/notifications/verify-hmac-signatures)

rikterbeek commented:
HI @danielduqma ,

Thanks for the changes we will do a final validation before we will merge it into the code but so far this looks great!

Regards,
Rik
Adyen
